django/db/models/sql/query.py

```diff
@@ -453,6 +453,9 @@ def get_aggregation(self, using, aggregate_exprs):
                     # filtering against window functions is involved as it
                     # requires complex realising.
                     annotation_mask = set()
+                    if isinstance(self.group_by, tuple):
```

Comment:
By running the aggregation related tests (`aggregation aggregation_regress queries expressions`) I've confirmed that the only other case where this branch is reached with an empty `annotation_mask` was from `aggregation_regress.tests.AggregationTests.test_more_more_more`

https://github.com/django/django/blob/4bf4222010fd8e413963c6c873e4088614332ef9/tests/aggregation_regress/tests.py#L1071-L1075

Thing is this test group by a column so it ends up in the select clause even if no annotations a selected.

All that to say that we didn't have a single test that covered the case where we group solely by an annotation (not a column) and performed aggregation over it.